pkg/controller/operators/catalog/subscription/syncer.go

```diff
@@ -70,6 +70,7 @@ func (s *subscriptionSyncer) Sync(ctx context.Context, event kubestate.ResourceE
 		initial = initial.Update()
 	case kubestate.ResourceDeleted:
 		initial = initial.Delete()
+		metrics.DeleteSubsMetric(res)
```

Comment:
This only deletes the time series when the subscription has been deleted, but from the bug report:

> Metrics for Subscriptions of CSVs that have been upgraded to new versions still hang around after new Subscription/CSV have succeeded. 

Because installedCSV is one of the labels/values, we would need to delete the metric (with any non-current installedCSV) on upgrade as well. Or have I misunderstood the bug?

Reply:
@ecordell removing the metrics for the old subscription when a subscription was the original intention of the bug. 

Updated the PR.